src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Data/DataBindEngine.cs

```diff
@@ -368,13 +368,14 @@ internal ViewRecord GetViewRecord(object collection, CollectionViewSource key, T
                                                         Type targetType,
                                                         bool targetToSource)
         {
-            IValueConverter result = _valueConverterTable[sourceType, targetType, targetToSource];
-
-            if (result == null)
+            ValueConverterTableKey key = new ValueConverterTableKey(sourceType, targetType, targetToSource);
+            if (_valueConverterTable.TryGetValue(key, out IValueConverter result))
```

Comment:
Shouldn't this be
```suggestion
            if (!_valueConverterTable.TryGetValue(key, out IValueConverter result))
```
?
```suggestion
            if (_valueConverterTable.TryGetValue(key, out IValueConverter result))
```